src/Symfony/Component/HttpKernel/Tests/DependencyInjection/RegisterControllerArgumentLocatorsPassTest.php

```diff
@@ -524,26 +525,51 @@ public function testTaggedIteratorAndTaggedLocatorAttributes()
         /** @var ServiceLocator $locator */
         $locator = $container->get($locatorId)->get('foo::fooAction');
 
-        $this->assertCount(6, $locator->getProvidedServices());
+        $this->assertCount(2, $locator->getProvidedServices());
 
         $this->assertTrue($locator->has('iterator1'));
         $this->assertInstanceOf(RewindableGenerator::class, $argIterator = $locator->get('iterator1'));
         $this->assertCount(2, $argIterator);
 
-        $this->assertTrue($locator->has('iterator2'));
-        $this->assertInstanceOf(RewindableGenerator::class, $argIterator = $locator->get('iterator2'));
-        $this->assertCount(2, $argIterator);
-
         $this->assertTrue($locator->has('locator1'));
         $this->assertInstanceOf(ServiceLocator::class, $argLocator = $locator->get('locator1'));
         $this->assertCount(2, $argLocator);
         $this->assertTrue($argLocator->has('bar'));
         $this->assertTrue($argLocator->has('baz'));
 
         $this->assertSame(iterator_to_array($argIterator), [$argLocator->get('bar'), $argLocator->get('baz')]);
+    }
+
+    public function testAutowireIteratorAndAutowireLocatorAttributes()
```

Comment:
I've split the test in legacy and new attributes. There is no `expectDeprecation` so the  test passes with symfony/dependency-injection 7.0